siliconcompiler/schema/schema_cfg.py

```diff
@@ -215,6 +215,74 @@ def schema_cfg():
+
+###############################################################################
+# SCHEMATIC
+###############################################################################
+def schema_schematic(cfg):
+    ''' Schematic
+    '''
+
+    scparam(cfg, ['schematic', 'component'],
+            sctype='[(str,str)]',
+            shorthelp="Schematic: components",
+            switch="-schematic_component <(str,str)>",
+            example=["cli: -schematic_component (B0,NAND2X1)",
+                     "api: chip.set('schematic', 'component', ('B0','NAND2X1'))"],
+            schelp="""Components in the design, specified as a list of
+            (INSTANCE,MODEL) tuples.""")
+
+    scparam(cfg, ['schematic', 'pin'],
+            sctype='[(str,str)]',
+            shorthelp="Schematic: pins",
+            switch="-schematic_pin 'name <(str,str)>'",
+            example=["cli: -schematic_pin '(A,input)",
+                     "api: chip.set('schematic', 'pin', ('A','input'))"],
+            schelp="""Primary design I/O pins specified as a list of
+            (NAME,DIRECTION) tuples. Name can be any legal verilog string
+            name while DIRECTION is limited to input, output, and inout.""")
+
+    name = 'default'
+    scparam(cfg, ['schematic', 'net', name],
+            sctype='[str]',
+            shorthelp="Schematic: nets",
+            switch="-schematic_net 'name <str>'",
+            example=["cli: -schematic_net 'net0 I42.Z'",
+                     "api: chip.set('schematic', 'net', 'net0', 'I42.Z')"],
+            schelp="""Component and pin connectivity specified as a list
+            of connection points on a per net basis. The connection point
+            point format is "INSTANCE.PIN", where "." is the hierarchy
+            character. The "INSTANCE" field is optional for primary design
+            I/O pins. The "PIN" field is optional when connecting to
+            components with only one pin.""")
+
+    scparam(cfg, ['schematic', 'hierchar'],
+            sctype='str',
+            defvalue='.',
+            shorthelp="Schematic: hierarchy character",
+            switch="-schematic_hierchar <str>",
+            example=["cli: -schematic_hierchar '/'",
+                     "api: chip.set('schematic', 'hierchar', '/')"],
+            schelp="""Specifies the character used to express hierarchy. If
+            the hierarchy character is used as part of a name, it must be
+            escaped with a backslash('\').""")
+
+    scparam(cfg, ['schematic', 'buschar'],
+            sctype='str',
+            defvalue='[]',
+            shorthelp="Schematic: bus character",
+            switch="-schematic_buschar <str>",
+            example=["cli: -schematic_buschar '[]'",
+                     "api: chip.set('schematic', 'buschar', '[]')"],
+            schelp="""Specifies the character used to express bus bits. If the
+            bus character is used as part of a name, it must be
+            escaped with a backslash('\').""")
```

Comment:
do we really want to carry this around? I have yet to see anything other than []

Reply:
Will keep as discussed. Some weird schematic things floating around like <0>, <1> in some netlists.